### ISSUE TITLE
Panic when got disconnected from DB

### ISSUE THREAD

**Issue by Yonle** [barkyq/gnost-relay#9]
Gnost panic when got disconnected from DB.

![Screenshot_2023_0530_155834](https://github.com/barkyq/gnost-relay/assets/63401712/1587cd2a-02b2-4920-9f59-5066e38fbe25)

Consider making it reconnect to DB when possible.

**Comment by barkyq** [comment]
Yes good idea. You manually closed the DB to make it do this? Will add something to try to reconnect

**Comment by Yonle** [comment]
> Yes good idea. You manually closed the DB to make it do this? Will add something to try to reconnect

Sometime my PostgreSQL just got disconnected by itself.

**Comment by theLockesmith** [comment]
I was able to reproduce this once so far, but not consistently. Even if I kill the database entirely. Any idea how to reliably reproduce so I can test?

**Comment by theLockesmith** [comment]
I'm digging into this one and it looks like the error handling just needs to be adjusted if the connection ever comes back nil. The question then becomes, how many times do we let it come back nil before we **_do_** panic? It strikes me as a bad idea to let it retry indefinitely, especially on a node that sees any kind of public traffic.

How does something like that sound to you, @barkyq? The two errors I've managed to see are the if/else on line 557 and exec on line 502. If those panics get wrapped inside a few retries maybe that would solve intermittent connection issues, but eventually break out if the db becomes long term unreachable?

**Comment by barkyq** [comment]
That sounds good to me! Certainly this would be good to fix.

Sep 1, 2023 3:35:36 PM TheLockesmith ***@***.***>:
